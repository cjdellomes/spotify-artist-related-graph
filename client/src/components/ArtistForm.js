import React from 'react';

class ArtistForm extends React.Component {
    constructor(props) {
        super(props);

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleReset = this.handleReset.bind(this);
    }

    handleChange(event) {
        this.props.onArtistChange(event.target.value);
    }

    handleSubmit(event) {
        console.log('Artist submitted: ' + this.props.searchValue);
        this.props.onArtistSubmit(this.props.searchValue);
        event.preventDefault();
    }

    handleReset(event) {
        this.props.onGraphReset();
        event.preventDefault();
    }

    render() {
        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    <label>
                        Artist:
                        <input type="text" value={this.props.searchValue} onChange={this.handleChange} />
                    </label>
                    <input type="submit" onSubmit={this.handleSubmit} />
                    <input type="reset" onClick={this.handleReset} />
                </form>
            </div>
        );
    }
}

export default ArtistForm;