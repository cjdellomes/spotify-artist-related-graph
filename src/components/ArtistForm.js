import React from 'react';

class ArtistForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {value: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }   

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    handleSubmit(event) {
        console.log('Artist submitted: ' + this.state.value);
        event.preventDefault();
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <label>
                    Artist:
                    <input type="text" value={this.state.value} onChange={this.handleChange}/>
                </label>
                <input type="submit" onSubmit={this.handleSubmit}/>
            </form>
        );
    }
}

export default ArtistForm;