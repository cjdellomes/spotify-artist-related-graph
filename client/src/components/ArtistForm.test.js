import React from 'react';
import renderer from 'react-test-renderer';
import ArtistForm from './ArtistForm';

it('renders correctly', () => {
  const tree = renderer
    .create(<ArtistForm
      searchValue="test"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
