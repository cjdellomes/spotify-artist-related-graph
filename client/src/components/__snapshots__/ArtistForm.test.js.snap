// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="transparent-blur"
  id="form-box"
  onSubmit={[Function]}
>
  <button
    onClick={[Function]}
    type="button"
  >
    Reset
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
  <div
    id="search-field"
  >
    <input
      onChange={[Function]}
      placeholder="Artist Name"
      type="text"
      value="test"
    />
  </div>
</form>
`;
